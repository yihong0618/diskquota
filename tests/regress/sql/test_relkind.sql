CREATE DATABASE test_relkind;
\c test_relkind
CREATE TYPE test_type AS (
        "dbid" oid,
        "datname" text
);
CREATE VIEW v AS select * from pg_class;
CREATE EXTENSION diskquota;
CREATE table test(a int);
SELECT diskquota.init_table_size_table();
-- diskquota.table_size should not change after creating a new type
SELECT tableid::regclass, size, segid
FROM diskquota.table_size 
WHERE segid = -1 AND tableid::regclass::name NOT LIKE '%.%'
ORDER BY tableid;

SELECT diskquota.pause();
SELECT diskquota.wait_for_worker_new_epoch();
DROP EXTENSION diskquota;
\c contrib_regression
DROP DATABASE test_relkind;
