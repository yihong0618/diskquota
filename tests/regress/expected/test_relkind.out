CREATE DATABASE test_relkind;
\c test_relkind
CREATE TYPE test_type AS (
        "dbid" oid,
        "datname" text
);
CREATE VIEW v AS select * from pg_class;
CREATE EXTENSION diskquota;
CREATE table test(a int);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'a' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
SELECT diskquota.init_table_size_table();
 init_table_size_table 
-----------------------
 
(1 row)

-- diskquota.table_size should not change after creating a new type
SELECT tableid::regclass, size, segid
FROM diskquota.table_size 
WHERE segid = -1 AND tableid::regclass::name NOT LIKE '%.%'
ORDER BY tableid;
 tableid | size | segid 
---------+------+-------
 test    |    0 |    -1
(1 row)

SELECT diskquota.pause();
 pause 
-------
 
(1 row)

SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

DROP EXTENSION diskquota;
\c contrib_regression
DROP DATABASE test_relkind;
